space/space.go

```diff
@@ -295,9 +295,57 @@ func (r *GormRepository) Create(ctx context.Context, space *Space) (*Space, erro
 	return space, nil
 }
 
+// SortSpaceBy is type used to define parameters using which spaces will be sorted
+type SortSpaceBy string
+
+var (
+	SortSpaceByNameAsc       = SortSpaceBy("name ASC")
+	SortSpaceByNameDesc      = SortSpaceBy("name DESC")
+	SortSpaceByOwnerIDAsc    = SortSpaceBy("owner_id ASC")
+	SortSpaceByOwnerIDDesc   = SortSpaceBy("owner_id DESC")
+	SortSpaceByCreatedAtAsc  = SortSpaceBy("created_at ASC")
+	SortSpaceByCreatedAtDesc = SortSpaceBy("created_at DESC")
+	SortSpaceByUpdatedAtAsc  = SortSpaceBy("updated_at ASC")
+	SortSpaceByUpdatedAtDesc = SortSpaceBy("updated_at DESC")
+	SortSpaceByDefault       = SortSpaceByUpdatedAtDesc
+)
+
+// ParseSortSpaceBy parses the string input and returns object of type SortSpaceBy
+// which can directly be used while querying database to order the output.
+func ParseSortSpaceBy(s *string) (SortSpaceBy, error) {
+	if s == nil {
+		// this is the default case
+		// which returns recently updated spaces first
+		return SortSpaceByDefault, nil
+	}
+
+	var sort SortSpaceBy
+	switch *s {
+	case "name":
+		sort = SortSpaceByNameAsc
+	case "-name":
+		sort = SortSpaceByNameDesc
+	case "owner":
+		sort = SortSpaceByOwnerIDAsc
+	case "-owner":
+		sort = SortSpaceByOwnerIDDesc
+	case "created":
+		sort = SortSpaceByCreatedAtAsc
+	case "-created":
+		sort = SortSpaceByCreatedAtDesc
+	case "updated":
+		sort = SortSpaceByUpdatedAtAsc
+	case "-updated":
+		sort = SortSpaceByUpdatedAtDesc
+	default:
+		return SortSpaceBy(""), fmt.Errorf("unknown sort parameter: %s", *s)
```

Comment:
I would suggest to change return value as string pointer and error. Then return like this here:
`return nil, errors.NewBadParameterError("sort", *s)`
This will be converted to `400` http error status in the controller (see my comment above in controller).

Reply:
@baijum I agree with the bad parameter error but I'm skeptical if we need to return a pointer. Might as well be an empty string. The caller should not use the value anyway if `err` is not `nil`.